src/storm/modelchecker/helper/infinitehorizon/SparseDeterministicInfiniteHorizonHelper.cpp

```diff
@@ -349,10 +341,18 @@ std::vector<ValueType> SparseDeterministicInfiniteHorizonHelper<ValueType>::comp
 
 template<typename ValueType>
 std::vector<ValueType> SparseDeterministicInfiniteHorizonHelper<ValueType>::computeSteadyStateDistrForBsccEqSys(
-    Environment const& env, storm::storage::StronglyConnectedComponent const& bscc) {
+    Environment const& inputEnv, storm::storage::StronglyConnectedComponent const& bscc) {
     // We want that the returned values are sorted properly. Let's assert that strongly connected components use a sorted container type
     STORM_LOG_ASSERT(std::is_sorted(bscc.begin(), bscc.end()), "Expected that bsccs are sorted.");
 
+    // Prepare an environment for the underlying linear equation solver
+    auto env = inputEnv;
+    if (env.solver().getLinearEquationSolverType() == storm::solver::EquationSolverType::Topological) {
+        // Topological solver does not make any sense since the BSCC is connected.
+        env.solver().setLinearEquationSolverType(env.solver().topological().getUnderlyingEquationSolverType(),
+                                                 env.solver().topological().isUnderlyingEquationSolverTypeSetFromDefault());
```

Comment:
I dont understand this last call?

Reply:
Essentially, this disables topological optimizations by setting the current linear equation solver to the one that the topological solver would use to solve the SCCs. More specifically, we set the equation solver type and whether it has been set explicitly by the user (e.g. via `--topological:eqsolver native`).

Reply:
I get generally a bit confused by this; this means that as a user (via CLI or API) I set a method, but the code overrides it and selects another solver. I wonder whether a user can understand this (and whether they have to).

(I am a bit afraid for statements like "We set storm to use topological ... " when Storm than does not use this topological optimizations, for good reasons).

Reply:
I think we want the same thing. When a user enables topological optimizations, the intention should be to use topological optimizations whenever possible.

This was not the case with the previous code because we accidentally disabled topological solving in cases where it *could* have some benefit. 

With this PR, we "fix" that by overwriting the user input only in cases where topological optimizations are known to be impossible (because the considered states are known to be strongly connected).

From a user perspective, disabling topological solving at this point does not make any difference (except maybe a  performance improvement because we avoid an unnecessary graph analysis and a matrix copy)

To summarize, my aim here is to disable topological optimizations if **and only if** they would not have any positive effect.

Reply:
I dont disagree with this PR, but I have a more general discussion that we can have elsewhere :-)